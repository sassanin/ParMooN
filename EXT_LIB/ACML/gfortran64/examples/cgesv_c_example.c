
/* cgesv Example Program Text */
/*
 * ACML version 1.0 Copyright AMD,NAG 2003
 */

#include <acml.h>
#include <stdio.h>

int main(void)
{
#define NMAX 8
#define NRHMAX 8
  const int lda=NMAX, ldb=NMAX;
  int i, info, j, n, nrhs;
  complex a[NMAX*NMAX], b[NMAX*NRHMAX];
  int ipiv[NMAX];

  /* These macros allow access to 1-d arrays as though
     they are 2-d arrays stored in column-major order,
     as required by ACML C routines. */
#define A(I,J) a[((J)-1)*lda+(I)-1]
#define B(I,J) b[((J)-1)*ldb+(I)-1]

  printf("\n");
  printf("ACML example: solution of linear equations using cgesv\n");
  printf("------------------------------------------------------\n");
  printf("\n");

  /* Initialize matrix A */
  n = 4;
  A(1,1) = compose_complex(-1.34, 2.55);
  A(1,2) = compose_complex( 0.28, 3.17);
  A(1,3) = compose_complex(-6.39,-2.20);
  A(1,4) = compose_complex( 0.72,-0.92);
  A(2,1) = compose_complex(-0.17,-1.41);
  A(2,2) = compose_complex( 3.31,-0.15);
  A(2,3) = compose_complex(-0.15, 1.34);
  A(2,4) = compose_complex( 1.29, 1.38);
  A(3,1) = compose_complex(-3.29,-2.39);
  A(3,2) = compose_complex(-1.91, 4.42);
  A(3,3) = compose_complex(-0.14,-1.35);
  A(3,4) = compose_complex( 1.72, 1.35);
  A(4,1) = compose_complex( 2.41, 0.39);
  A(4,2) = compose_complex(-0.56, 1.47);
  A(4,3) = compose_complex(-0.83,-0.69);
  A(4,4) = compose_complex(-1.96, 0.67);

  /* Initialize right-hand-side matrix B */
  nrhs = 2;
  B(1,1) = compose_complex(26.98, 50.86);
  B(1,2) = compose_complex(31.32, -6.70);
  B(2,1) = compose_complex( 7.72, -7.30);
  B(2,2) = compose_complex(15.86, -1.42);
  B(3,1) = compose_complex(-4.03, 26.66);
  B(3,2) = compose_complex(-2.15, 30.19);
  B(4,1) = compose_complex(-0.80,  3.24);
  B(4,2) = compose_complex(-2.56,  7.55);

  printf("Matrix A:\n");
  for (i = 1; i <= n; i++)
    {
      for (j = 1; j <= n; j++)
        printf(" (%8.4f,%8.4f)", A(i,j).real, A(i,j).imag);
      printf("\n");
    }

  printf("\n");
  printf("Right-hand-side matrix B:\n");
  for (i = 1; i <= n; i++)
    {
      for (j = 1; j <= nrhs; j++)
        printf(" (%8.4f,%8.4f)", B(i,j).real, B(i,j).imag);
      printf("\n");
    }

  /* Factorize A */
  cgesv(n,nrhs,a,lda,ipiv,b,ldb,&info);

  printf("\n");
  if (info == 0)
    {
      /* Print solution */
      printf("Solution matrix X of equations A*X = B:\n");
      for (i = 1; i <= n; i++)
        {
          for (j = 1; j <= nrhs; j++)
            printf(" (%8.4f,%8.4f)", B(i,j).real, B(i,j).imag);
          printf("\n");
        }
    }
  else
    printf("The matrix A is singular\n");

  return 0;
}
