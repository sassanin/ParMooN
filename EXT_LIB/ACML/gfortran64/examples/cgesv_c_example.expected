
ACML example: solution of linear equations using cgesv
------------------------------------------------------

Matrix A:
 ( -1.3400,  2.5500) (  0.2800,  3.1700) ( -6.3900, -2.2000) (  0.7200, -0.9200)
 ( -0.1700, -1.4100) (  3.3100, -0.1500) ( -0.1500,  1.3400) (  1.2900,  1.3800)
 ( -3.2900, -2.3900) ( -1.9100,  4.4200) ( -0.1400, -1.3500) (  1.7200,  1.3500)
 (  2.4100,  0.3900) ( -0.5600,  1.4700) ( -0.8300, -0.6900) ( -1.9600,  0.6700)

Right-hand-side matrix B:
 ( 26.9800, 50.8600) ( 31.3200, -6.7000)
 (  7.7200, -7.3000) ( 15.8600, -1.4200)
 ( -4.0300, 26.6600) ( -2.1500, 30.1900)
 ( -0.8000,  3.2400) ( -2.5600,  7.5500)

Solution matrix X of equations A*X = B:
 (  1.0000,  1.0000) ( -1.0000, -2.0000)
 (  2.0000, -3.0000) (  5.0000,  1.0000)
 ( -4.0000, -5.0000) ( -3.0000,  4.0000)
 (  1.0000,  6.0000) (  2.0000, -3.0000)
